### DISCUSSION TITLE
alpine.reactive how to use globally

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#3902] [4. General]
![image](https://github.com/alpinejs/alpine/assets/9047436/dd036a57-eca1-499f-a9d1-b96ed628f541)

Hi,

I want to do some experiment, I am very new to alpine, but I like it.
I want to do global reactive and effect experiment, but I have no idea why it does not work.

I use a button in some small x-data scope to click trigger event to make data.count = 2, and console warned me data.count is not defined.

Thanks
Eric Xin

**Comment by ericxin1982** [comment]
<button @click="data.count = 2">Fire</button>

![image](https://github.com/alpinejs/alpine/assets/9047436/f3a9886a-ffc1-469d-9a59-1d314b306ee9)

I try to understand how global reactive and effect works.

Thanks
Eric Xin

**Comment by SimoTod** [comment]
`let data` inside a closure is not a global variable. Let makes your variable available only in the block were it's defined. In general, variables scoped globally are not a great idea since other scripts can override them without you realising it, leading to bugs hard to reproduce

**Comment by ericxin1982** [comment]
Ok.

**Comment by ekwoka** [comment]
If you must, you'd need to do `window.data = Alpine.reactive({ count: 0 })`

But you might also just want to use a store

since stores are available globally..

**Reply by ericxin1982** [reply]
Thanks
